libraries/GCS_MAVLink/GCS_Common.cpp

```diff
@@ -1340,7 +1347,9 @@ void GCS::retry_deferred()
             chan(i).retry_deferred();
         }
     }
+    _statustext_sem->take_blocking();
     service_statustext();
```

Comment:
Some comment on service_statustext indicating it must be called with `_statustext_sem`, perhaps.

Reply:
fixed